cmd/restic/cmd_recover.go

```diff
@@ -24,15 +25,29 @@ Exit status is 0 if the command was successful, and non-zero if there was any er
 `,
 	DisableAutoGenTag: true,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		return runRecover(globalOptions)
+		return runRecover(globalOptions, recoverOptions)
 	},
 }
 
+type RecoverOptions struct {
+	All     bool
+	Combine bool
+	DryRun  bool
+	Tags    []string
+}
+
+var recoverOptions RecoverOptions
+
 func init() {
 	cmdRoot.AddCommand(cmdRecover)
+	recoverFlags := cmdRecover.Flags()
+	recoverFlags.BoolVar(&recoverOptions.All, "all", false, "add all directories, not only unreferenced ones")
+	recoverFlags.BoolVar(&recoverOptions.Combine, "combine", true, "if true, combine all directories into one snapshot, else generate multple snapshots")
```

Comment:
With the new prune implementation I'd expect there to be thousands of unreferenced trees, so I doubt that this option would still be useful. A option which must be passed as `--combine=false` is really confusing.

Reply:
Thanks for the hint about the pruning. This applies to `recover` with and without combine. The only difference is whether the unreferenced trees are combined in one new snapshot (as current `recover` does) or in subtrees within one single new snapshot. I would always prefer to use `--combine=false` as this allows to clean up the unneeded snapshots.
But as you can specify tags for the "uncombined" snapshots, even lots of snapshots could be handled or easily removed if not needed.

About the option name: What about inverting it: `--split` with default "false"?

Reply:
About the unreferenced trees: I made #3228 and think we should merge this in order to save the `recover` command (independent from this PR).

I already changed the name to `--split`.